src/smc-hub/client.coffee

```diff
@@ -2114,6 +2136,48 @@ class exports.Client extends EventEmitter
                 else
                     @push_to_client(message.stripe_subscriptions(id:mesg.id, subscriptions:subscriptions))
 
+    mesg_stripe_get_coupon: (mesg) =>
+        dbg = @dbg("mesg_stripe_get_coupon")
+        dbg("get the coupon with id == #{mesg.coupon_id}")
+        if not @ensure_fields(mesg, 'coupon_id')
+            dbg("missing field coupon_id")
+            return
+        @validate_coupon mesg.coupon_id, (err, coupon) =>
+            if err
+                @stripe_error_to_client(id:mesg.id, error:err)
+            else
+                @push_to_client(message.stripe_coupon(id:mesg.id, coupon:coupon))
+
+    # Checks these coupon criteria:
+    # - Exists
+    # - Is valid
+    # - Used by this account less than the max per account (hard coded default is 1)
+    # Calls cb(err, coupon, coupon_history)
+    validate_coupon: (coupon_id, cb) =>
+        dbg = @dbg("validate_coupon")
+        @_stripe = get_stripe()
+        async.series([
+            (local_cb) =>
```

Comment:
So you know, if you just call this (cb) rather than (local_cb) there is no problem whatever with scoping.  It does NOT conflict with the cb in lie 2156 above. I've done this about 10K times, at least....

Reply:
(not saying you should change anything -- just letting you know in case)

Reply:
I know. I just hate reading it when it's a long function (it was a long function during writing).